contract-message-predicate/src/contract_message_test.sw

```diff
@@ -16,6 +12,31 @@ storage {
     data4: Address = Address::from(ZERO_B256),
 }
 
+// Define verification abi
+abi VerifyMessageData {
+    #[storage(read)]
+    fn get_test_counter() -> u64;
+    #[storage(read)]
+    fn get_test_data1() -> ContractId;
+    #[storage(read)]
+    fn get_test_data2() -> u64;
+    #[storage(read)]
+    fn get_test_data3() -> b256;
+    #[storage(read)]
+    fn get_test_data4() -> Address;
+}
+
+// Get the length of a message input data
+pub fn input_message_data_length(index: u64) -> u64 {
+    __gtf::<u64>(index, GTF_INPUT_MESSAGE_DATA_LENGTH)
+}
+
+// Get the data of a message input
+pub fn input_message_data<T>(index: u64, offset: u64) -> T {
```

Comment:
Same with this one, though it now returns a `Bytes` type. I refactored the Bridge-fungible-token contract to handle the new return type already [here](https://github.com/FuelLabs/bridge-fungible-token/blob/f214595fead2b40820a117a49248ae4465ff11d3/bridge-fungible-token/src/utils.sw#L144).

Reply:
How can I convert the Bytes to a u64?

Reply:
Hmm, we'd have to build a local conversion function based on the into() method for Bytes that converts to b256.
Ideally we would just impl TryInto <u64>() for Bytes, but we don't have that trait yet (blocked).

Reply:
Bytes to b256 is already supported for `into` [see here](https://github.com/FuelLabs/sway/blob/9934103647eec6eef2db5b56cf51f62a8984d796/sway-lib-std/src/b256.sw).

Reply:
Eventually it should be added via implementing TryInto<Bytes> for u64, but this can't be done yet. The workaround is to impl From<u64> for Bytes and use the `into()`method, but this is also blocked atm.